integration_tests/src/main/python/parquet_pyarrow_test.py

```diff
@@ -0,0 +1,255 @@
+    else:
+        return Row(**py_dict)
+
+
+def _convert_pyarrow_list_object_to_cpu_object(pyarrow_list_scalar):
+    """Convert pyarrow list to python list"""
+    # pyarrow.ListScalar obj to python list
+    py_list = pyarrow_list_scalar.as_py()
+    if py_list is None:
+        return None
+    else:
+        return py_list
+
+
+def _convert_pyarrow_map_object_to_cpu_object(pyarrow_map_scalar):
+    """Convert pyarrow map to python dict"""
+    # pyarrow.MapScalar obj to python (key, value) pair list
+    py_key_value_pair_list = pyarrow_map_scalar.as_py()
+    if py_key_value_pair_list is None:
+        return None
+    else:
+        return dict(py_key_value_pair_list)
+
+
+def _convert_pyarrow_column_to_cpu_object(pyarrow_column, pa_type):
+    """ convert pyarrow column to a column contains CPU type objects"""
+    if isinstance(pa_type, pa.StructType):
+        # struct column
+        return [_convert_pyarrow_struct_object_to_cpu_object(v) for v in pyarrow_column]
+    elif isinstance(pa_type, pa.ListType):
+        # list column
+        return [_convert_pyarrow_list_object_to_cpu_object(v) for v in pyarrow_column]
+    elif isinstance(pa_type, pa.MapType):
+        # map column
+        return [_convert_pyarrow_map_object_to_cpu_object(v) for v in pyarrow_column]
+    else:
+        # other type column
+        return [v.as_py() for v in pyarrow_column]
+
+
+def _gen_row_from_columnar_cpu_data(column_names, columnar_data, row_idx):
+    """Gen a Pyspark Row from columnar data for a row index"""
+    key_value_pair_list = [(column_names[col], columnar_data[col][row_idx]) for col in range(len(column_names))]
+    return Row(**dict(key_value_pair_list))
+
+
+def _convert_pyarrow_table_to_cpu_object(pa_table):
+    """
+    Convert pyarrow table to cpu type data to reuse the `_assert_equal(cpu, gpu)` in asserts.py
+     e.g.: convert pyarrow map_scalar to Pyspark Row
+    """
+    columnar_data = []
+    for pa_column, pa_type in zip(pa_table.columns, pa_table.schema.types):
+        converted_column = _convert_pyarrow_column_to_cpu_object(pa_column, pa_type)
+        columnar_data.append(converted_column)
+    cpu_data = [
+        _gen_row_from_columnar_cpu_data(pa_table.schema.names, columnar_data, row_idx)
+        for row_idx in range(pa_table.num_rows)
+    ]
+    return cpu_data
+
+
+def assert_equal(cpu_adapted_from_pyarrow, gpu):
+    """Verify that the result from the pyarrow(already adapted to CPU data format) and the GPU are equal"""
+    try:
+        _assert_equal(cpu_adapted_from_pyarrow, gpu, float_check=get_float_check(), path=[],
+                      cpu_data_adapted_from="pyarrow")
+    except:
+        sys.stdout.writelines(difflib.unified_diff(
+            a=[f"{x}\n" for x in cpu_adapted_from_pyarrow],
+            b=[f"{x}\n" for x in gpu],
+            fromfile='pyarrow OUTPUT',
+            tofile='GPU OUTPUT'))
+        raise
+
+
+def assert_gpu_and_pyarrow_are_compatible(base_write_path, gen_list, conf={}):
```

Comment:
My main concern here is the path. In our tests we often rely on the default HDFS path for testing so we don't have a fully qualified path/URL like `s3a://...` that is fine because CPU vs GPU both use the same FileSystem abstractions so we both point to the same place. But here we don't have those same abstractions so if the path is not a local file path these tests are likely to fail. We should either have a way to fix that for cases like dataproc where we test on a full cluster, or we need to have a good way to detect it and skip the tests. Making it so the tests run is preferable.

Reply:
Will add an option to disable the cases.

Reply:
Done.